source/randLIB.c

```diff
@@ -90,26 +91,23 @@ void randLIB_seed_random(void)
         random_file = fopen(RANDOM_DEVICE, "rb");
     }
 #else
-    arm_random_module_init();
-
     /* We exclusive-OR with the current state, in case they make this call
      * multiple times,or in case someone has called randLIB_add_seed before
      * this. We don't want to potentially lose entropy.
      */
 
+    srand(time(0));
     /* Spell out expressions so we get known ordering of 4 seed calls */
-    uint64_t s = (uint64_t) arm_random_seed_get() << 32;
-    state[0] ^= (s | arm_random_seed_get());
+    uint64_t s = (uint64_t) rand() << 32;
```

Comment:
How are embedded devices getting a random seed now? this needs to have even minute randomness between devices to allow them to work?

Is there some other way you are adding randomness to devices?

Reply:
Yes This is not as important as the add seed is called in all places that can add entropy and all stacks seems to do it

Reply:
Missed this, but I would have put a big "X" on it. I understand why you'd want to remove the "entropy pull" and rely on the app/port pushing in entropy, but

1. This introduces a new dependency on C library rand(), defeating one point of the library
2. It disrupts the seed of C library rand - the app may be intentionally wanting a seeded random for its own test purposes
3. It doesn't check whether the app IS actually pushing in entropy.

If going on this path I would prefer to see an assert that at least one call to the "add entropy" call has been made before asking for a random number.

Reply:
An alternative that eliminates part of my objections, and works better, is to replace the `rand` and `srand` with just using `splitmix64`.  Or, more simply, just have init do `randLIB_add_seed(time(NULL))`.

But I still object to that due to pulling in platform RTC code, and you don't know it's going to actually randomise.

Reply:
https://github.com/PelionIoT/mbed-client-randlib/pull/17